### DISCUSSION TITLE
Prisma MongoDB Serverless Instance Charging Too Much Money For Queries.

### DISCUSSION THREAD

**Discussion by casperibo** [prisma/prisma#13900] [Q&A] [answered]
Hi, I have Prisma MongoDB project and I'm using MongoDB Atlass Serverless Instance.

As I know serverless instances charging per read queries($0.10/1M reads).

I think my "prisma.findUnique" queries charging too much money because It's scan every document to find the matching unique value.

I have "Notes" collection and every note item has unique "slug" field.

Also every "User" collection has unique "username" field, I think when visitor try to visit a user's profile, the "prisma.findUnique" query scanning every document so It's mean a lot of reading process for MongoDB Serverless Instance.

How can I decrease my document reads?

I will be appreciate it for every suggestion or help.

**Comment by austincrim** [comment, marked as answer]
Hey @casperibo 👋,

Have you measured your document reads against your actual traffic? How does that compare?

You can also enable [query logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#log-to-stdout) to see what kind of document reads Prisma is generating. Maybe we can find somewhere to go from there 👍.

**Reply by casperibo** [reply]
Hi @austincrim thank you for your response.

I take a look my documents reads right now and I took average daily reads by looking last 7 days.

My daily document reads is 27 million RPU.

And my daily visitors around 400-500.

Before I re create my project, my project was using AWS Amplify and I was paying around $15 USD.

After I re create my project with Apollo, Prisma, MongoDB Serverless Instance, my last MongoDB Serverless Instance bill was $55 USD and It's keep increasing.

I can try to make query logging but I'm not sure I can.

I believe my findUnique queries causing the billing because let's think I have 500 user and every user has unique username. If one visitor is visiting any user, I guess MongoDB try to read every users until find the unique username.

I can't imagine how many documents reads 500 people visiting any user's profile.

I think one "getUserByUsername" query causing 0-500 documents reads for each visitors.

**Reply by austincrim** [reply]
My colleague suggested that you double-check the indices on your collection. You should have a unique index for the `username` field, but it's possible something got out of sync.

**Reply by casperibo** [reply]
Hi again @austincrim I really appreciate it for your valuable answers.

I'm using unique index for username, maybe I'm wrong maybe other things causes the cost, I think I have to do log thing that you suggest.

Also I want to share my prisma informations, maybe you recognize something.

```
generator client {
  provider = "prisma-client-js"
}
```

I don't have preview features line in my generator client {}

`previewFeatures = ["mongoDb"]`

My related package versions:

```
"@prisma/client": "^3.13.0",
"prisma": "^3.13.0",
"@apollo/client": "3.5.10",
"apollo-server-micro": "^3.6.7",
"nexus": "^1.3.0",
"react": "17.0.2",
"next": "12.1.0",
```

Do you suggest me to update prisma latest version?

**Reply by janpio** [reply]
Always. The update should be pretty painless.

(I would not expect the read behavior of MongoDB to be impacted by that change - but you never know)

From your description this sounds to be isolated to 2 collections and queries for these. Can you maybe share the full schema of these two and the queries you are using? Also some information about the data in your database? That might help us understand what is going on here. Thanks.

**Comment by casperibo** [comment]
I stopped using MongoDB Serverless Instances until I figure it out how I build serverless instance friendly queries. I will try serverless instances for my next projects, thank you to you all who tried to help.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
